### PR TITLE
Fix squash output unscaling when using gSDE

### LINKED ISSUE #1592: [Question] Why does unscaling action behaves differently in training and eval
### ❓ Question

I'm using PPO with `squash_output=True` option. It looks statistics (e.g., average reward, episode length) differs during collecting rollout and evaluating policy. After digging down, I found this is caused by the different behavior of unscaling action in `collect_rollouts` and  `predict` function used while train and eval, respectively. Why is `unscale_action` not applied in `collect_rollouts()`?

https://github.com/DLR-RM/stable-baselines3/blob/d68ff2e17f2f823e6f48d9eb9cee28ca563a2554/stable_baselines3/common/policies.py#L352-L361

https://github.com/DLR-RM/stable-baselines3/blob/d68ff2e17f2f823e6f48d9eb9cee28ca563a2554/stable_baselines3/common/on_policy_algorithm.py#L174-L177

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

### PR DESCRIPTION
**Pull request by PatrickHelm**
## Description

Closes #1592 
1. Prevent setting `squash_output=True` if not `use_sde=True.`
2. Unscales actions taken during training if `squash_output=True.`

To check whether the actions have been unscaled appropriately, I had to copy the code of the Pendulum environment from Gymnasium into the test file and track the actions passed to `step(action)` there. 

While working on 1., the following question came to my mind: 
Why should `squash_output=True` not be allowed for regular PPO (for example) after all? 
I think there may be use cases where a `SquashedDiagGaussianDistribution` is more appropriate than a `DiagGaussianDistribution` with clipping. Do you agree? (this is related to #1540)

## Motivation and Context

closes #1592 
- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by PatrickHelm** [comment]
I formatted the Gymnasium code according to the guidelines. The CI workflow should run now.

File: stable_baselines3/common/on_policy_algorithm.py [hunk 1]
```diff
@@ -171,9 +171,13 @@ def collect_rollouts(
 
             # Rescale and perform action
             clipped_actions = actions
+            # Unsquash the actions (if perviously squashed), otherwise
             # Clip the actions to avoid out of bound error
             if isinstance(self.action_space, spaces.Box):
-                clipped_actions = np.clip(actions, self.action_space.low, self.action_space.high)
+                if self.policy.squash_output:
```
Comment (araffin) [inline stable_baselines3/common/on_policy_algorithm.py:177]: missing comment and the comment above should be moved down

File: stable_baselines3/common/policies.py [hunk 2]
```diff
@@ -489,6 +489,8 @@ def __init__(
 
         self.log_std_init = log_std_init
         dist_kwargs = None
+
+        assert not (squash_output and not use_sde), "squash_output is only available when using gSDE"
```
Comment (araffin) [inline stable_baselines3/common/policies.py:493]:
```suggestion
        assert not (squash_output and not use_sde), "squash_output=True is only available when using gSDE (use_sde=True)"
```

File: tests/test_sde.py [hunk 3]
```diff
@@ -61,18 +69,196 @@ def test_sde_check():
...
+        if buffer.actions.max() > 0.5:
+            assert np.max(env.actions) > 1.0
+        if buffer.actions.max() < -0.5:
+            assert np.min(env.actions) < -1.0
     model.policy.reset_noise()
     if model_class == SAC:
         model.policy.actor.get_std()
+
+
+# default pendulum env from gymnasium with action tracking
```
Comment (araffin) [inline tests/test_sde.py:103]: you don't need to redefine the full env for that, a wrapper will do

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Pull request merged**